core/src/main/java/org/apache/iceberg/expressions/ExpressionParser.java

```diff
@@ -0,0 +1,334 @@
+    generator.writeFieldName(LEFT_OPERAND);
+    toJson(expression.left(), generator);
+    generator.writeFieldName(RIGHT_OPERAND);
+    toJson(expression.right(), generator);
+    generator.writeEndObject();
+  }
+
+  public static void toJson(Or expression, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeStringField(TYPE, OR);
+    generator.writeFieldName(LEFT_OPERAND);
+    toJson(expression.left(), generator);
+    generator.writeFieldName(RIGHT_OPERAND);
+    toJson(expression.right(), generator);
+    generator.writeEndObject();
+  }
+
+  public static void toJson(Not expression, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeStringField(TYPE, NOT);
+    generator.writeFieldName(OPERAND);
+    toJson(expression.child(), generator);
+    generator.writeEndObject();
+  }
+
+  public static void toJson(True expression, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeStringField(TYPE, TRUE);
+    generator.writeEndObject();
+  }
+
+  public static void toJson(False expression, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeStringField(TYPE, FALSE);
+    generator.writeEndObject();
+  }
+
+  public static void toJson(Predicate<?, ?> predicate, JsonGenerator generator) throws IOException {
+    if (predicate instanceof UnboundPredicate) {
+      toJson((UnboundPredicate<?>) predicate, generator);
+    } else {
+      throw new IllegalArgumentException("Cannot convert predicate " + predicate);
+    }
+  }
+
+  public static void toJson(UnboundPredicate<?> predicate, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeStringField(TYPE, UNBOUNDED_PREDICATE);
+    generator.writeStringField(OPERATION, predicate.op().name().toLowerCase());
+    generator.writeFieldName(TERM);
+    toJson(predicate.term(), generator);
+    if (!(ONE_INPUTS.contains(predicate.op()))) {
+      generator.writeFieldName(LITERALS);
+      generator.writeStartArray();
+      for (Literal<?> literal : predicate.literals()) {
+        toJson(literal, generator);
+      }
+      generator.writeEndArray();
+    }
+
+    generator.writeEndObject();
+  }
+
+  public static void toJson(Term term, JsonGenerator generator) throws IOException {
+    if (term instanceof NamedReference) {
+      toJson((NamedReference<?>) term, generator);
+    } else {
+      throw new IllegalArgumentException("Cannot convert term " + term);
+    }
+  }
+
+  public static void toJson(NamedReference<?> term, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeStringField(TYPE, NAMED_REFERENCE);
+    generator.writeStringField(VALUE, term.name());
+    generator.writeEndObject();
+  }
+
+  public static void toJson(List<Literal<?>> literals, JsonGenerator generator) throws IOException {
```

Comment:
this method seems unnecessary, can be combined to parent method, also can use enhanced for loop